### DISCUSSION TITLE
Why do I sometimes have to sign in twice using credentials and a custom sign in page?

### DISCUSSION THREAD

**Discussion by malssid** [nextauthjs/next-auth#1964] [Help]
I have a custom sign in page and for some reason I keep having to sign in twice.

For example the first time I sign in it will send me to this url:
**http://localhost:3000/signin?callbackUrl=http://localhost:3000**
then when I sign in again it will successfully bring me to the authenticated home page. 

On my home page I have a getServerSideProps with 
`if (!session) {
    return {
      redirect: {
        destination: "/signin",
        permanent: false,
      },
    };
  }`
which redirects them to sign in if there is no session. I don't think this is the issue because if I go straight to localhost:3000/signin it still makes me sign in twice. One thing though is that on this signin page, if I full refresh once, it won't make me sign in twice like usual. 

Anyone know what could be wrong here? Let me know if you want to see more of my code. Btw I am using the form method instead of signIn();

**Comment by thomasmost** [comment]
I believe this is duped in #2426

This fixed it for me (in prod, no less!) https://github.com/nextauthjs/next-auth/issues/2426#issuecomment-1141406105

**Comment by risingblock** [comment]
I am still experiencing this problem using both the signIn() and the form method with a custom sign in page. 

I've tried all the solutions in #2426 but haven't found a solution.

**Comment by haruaki07** [comment]
This is just workaround, changed `NEXTAUTH_URL` environment variable to `http://` fixes this issue. 
For example: 

```
NEXTAUTH_URL=http://example.com
```

Haven't try this https://github.com/nextauthjs/next-auth/issues/2426#issuecomment-1141406105
